src/run/run_environment/local/provider.rs

```diff
@@ -17,85 +18,91 @@ use crate::run::{
 #[derive(Debug)]
 pub struct LocalProvider {
     repository_provider: RepositoryProvider,
+    owner: String,
+    repository: String,
     pub ref_: String,
-    pub owner: String,
-    pub repository: String,
     pub head_ref: Option<String>,
     pub base_ref: Option<String>,
     pub event: RunEvent,
     pub repository_root_path: String,
 }
 
-impl LocalProvider {}
-
-fn extract_provider_owner_and_repository_from_remote_url(
-    remote_url: &str,
-) -> Result<(RepositoryProvider, String, String)> {
-    let GitRemote {
-        domain,
-        owner,
-        repository,
-    } = parse_git_remote(remote_url)?;
-    let repository_provider = match domain.as_str() {
-        "github.com" => RepositoryProvider::GitHub,
-        "gitlab.com" => RepositoryProvider::GitLab,
-        domain => bail!(
-            "Repository provider {} is not supported by CodSpeed",
-            domain
-        ),
-    };
-
-    Ok((
-        repository_provider,
-        owner.to_string(),
-        repository.to_string(),
-    ))
-}
-
 impl TryFrom<&Config> for LocalProvider {
     type Error = Error;
-    fn try_from(_config: &Config) -> Result<Self> {
-        let repository_root_path = match find_repository_root(&std::env::current_dir()?) {
+    fn try_from(config: &Config) -> Result<Self> {
+        let current_dir = std::env::current_dir()?;
+        match find_repository_root(&current_dir) {
```

Comment:
This logic is a bit weird, if we are overriding the repository, we do not care about the actual repository state.

Let's simplify and branch directly on whether or not the repository is overridden, keeping the previous behavior is it is not.

Reply:
You will then be able to a test case for this, since there are no git repo repository involved

Reply:
Are you sure we don't want to still get git info if it's available ?

The use case I am thinking about is when someone forks a repo, and wants to run the benches on the fork/upstream explicitely without relying on the `origin` remote

Reply:
Since we have not specified this behavior yet, I would prefer that we implement what we need at the moment.
Depending on what we end up specing for forks and local runs, the behavior you implemented is bound to change anyway.